### ISSUE TITLE
Startup script error message is obnoxious on SSH

### ISSUE THREAD

**Issue by ammario** [coder/coder#13580]
My workspace personalize script often fails in a way that doesn't invalidate my development environment. Unfortunately, all `ssh` operations are met with a giant multi-page dump like this:

```
& rsync /tmp/sail coder.c0:/tmp/
👋 Your workspace is outdated! Update it here: https://dev.coder.com/@ammar/c0

==> ⧗ Running workspace agent startup scripts
2024-06-14 14:16:30.318-05:00 Startup Script: + trap 'touch /tmp/.coder-startup-script.done' EXIT
2024-06-14 14:16:30.318-05:00 Startup Script: + sudo service docker start
2024-06-14 14:16:30.366-05:00 Coder Login: Logging into Coder...
2024-06-14 14:16:30.366-05:00 Coder Login: 
2024-06-14 14:16:30.410-05:00 File Browser: Installing filebrowser 
2024-06-14 14:16:30.410-05:00 File Browser: 
2024-06-14 14:16:30.455-05:00 Git Clone: /home/coder/coder already exists and isn't empty, skipping clone!
2024-06-14 14:16:30.499-05:00 File Browser: Downloading File Browser for linux/amd64...
2024-06-14 14:16:30.542-05:00 code-server: Installing code-server!
2024-06-14 14:16:30.586-05:00 Personalize: ~ ~/coder
2024-06-14 14:16:30.586-05:00 Personalize: Hit:1 https://download.docker.com/linux/ubuntu jammy InRelease
...
```

upon failure. It's extremely annoying and unhelpful. Instead, we should just show a single warning line that suggests running a different command to see the full error.

**Comment by ammario** [comment]
This "bug" has been extremely terrible for me because my personalize script is many pages of logs. Also all of the logs are bolded

![Image](https://github.com/user-attachments/assets/9b34f9a3-2d08-4d18-8523-02ec4c242f78)

all in all a bad time

**Comment by ammario** [comment]
cc @mafredri — seems like an area of the codebase you're most familiar with.

**Comment by mafredri** [comment]
Step 1 could be to make sure we don't output this in non-interactive scenarios like `rsync`, I thought that was the case but perhaps never was or it changed at some point.

There's also a couple of other improvements that could be made like:
- Only output lines from the script that failed
- Only output last N lines (5-10?)

I agree the wall of text creates a pretty bad experience. It even slows down entry into workspace. Combining the aforementioned improvements and adding your suggestion for showing how to get the full log would IMO be a pretty good solution.

**Comment by ammario** [comment]
My preference is just two normal weight red lines that say your startup script failed, your workspace may not be complete, here's a command to see the logs and leave it at that.

**Comment by bpmct** [comment]
I agree with Ammar's comments here. Good to do 1-2 sentences explaining it failed and link / CLI instructions to view logs
